subsys/fs/nffs_fs.c

```diff
@@ -537,4 +540,47 @@ static int fs_init(struct device *dev)
 	return 0;
 }
 
+/* File operations */
+static const fs_file_ops_t nffs_file_ops = {
+	.open = nffs_open,
+	.close = nffs_close,
+	.read = nffs_read,
+	.write = nffs_write,
+	.lseek = nffs_seek,
+	.tell = nffs_tell,
+	.truncate = nffs_truncate,
+	.sync = nffs_sync,
+};
+
+/* Directory operations */
+static const fs_dir_ops_t nffs_dir_ops = {
+	.opendir = nffs_opendir,
+	.readdir = nffs_readdir,
+	.closedir = nffs_closedir,
+};
+
+/* Filesystem level operations */
+static const fs_file_system_ops_t nffs_fs_ops = {
+	.mount = nffs_mount,
+	.unlink = nffs_unlink,
+	.mkdir = nffs_mkdir,
+	.stat = nffs_stat,
+	.statvfs = nffs_statvfs,
+
+};
+
+/* Fileystsem info */
+static const fs_file_system_t nffs_fs = {
+	.fops = &nffs_file_ops,
+	.dops = &nffs_dir_ops,
+	.fs_ops = &nffs_fs_ops,
+};
+
+static int fs_init(struct device *dev)
```

Comment:
name: nffs_init, use it in SYS_INIT macro bellow

Reply:
Ok, will fix it.

Reply:
@nvlsianpu , I can add the fs_rename() API and corresponding VFS changes. You can add the nffs implementation. Is that sound good?